test/test_overrides.py

```diff
@@ -1270,15 +1270,34 @@ def __torch_function__(self, func, _, args=(), kwargs=None):
 
         self.assertEqual(out, ["layer2", "layer1"])
 
-    def test_error_using_same_mode(self):
+    def test_with_calls_restore_same_mode(self):
+        out = []
+
+        class A(TorchFunctionMode):
+            def __init__(self, msg):
+                self.msg = msg
+
+            def __torch_function__(self, func, _, args=(), kwargs=None):
+                if kwargs is None:
+                    kwargs = {}
+                out.append(self.msg)
+                return func(*args, **kwargs)
+
+        with A("A") as x:
+            with x:
```

Comment:
This looks a little weird here but is actually a pretty common occurrence (for an admittedly small amount of users) where because of function calls and such a user's not sure if they're in a mode or not but now definitely want to be in that mode. So it typically looks less silly than this